pkgs/test/stdenv/no-broken-symlinks.nix

```
@@ -0,0 +1,123 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+}:
+
+let
+  inherit (lib.strings) concatStringsSep;
+  inherit (pkgs) runCommand;
+  inherit (pkgs.testers) testBuildFailure;
+
+  mkDanglingSymlink = ''
+    ln -sr "$out/dangling" "$out/dangling-symlink"
+  '';
```

Let's test both the relative and absolute cases.

```suggestion
  mkDanglingSymlink = absolute: ''
    ln -sr "${if absolute then "$out/" else ""}dangling" "$out/dangling-symlink"
  '';
```